File: gin-backend/pkg/common/models/company_detail.go
package models

import "gorm.io/gorm"

type Company_detail struct {
	gorm.Model
	Comp_det_stock         int
	Comp_det_maximum_mount int
	Comp_det_minimun_moun  int
	Product_id             *uint
	Company_id             *uint
	Product                *Product
	Company                *Company
}

File: angular-frontend/src/app/demo/api/company_detail.ts
import { Company } from "./company"
import { Product } from "./products"

export interface Company_detail {
    ID?: number,
    Comp_det_stock?: number
    Comp_det_maximum_mount?: number
    Comp_det_minimun_moun?: number
    Product_id?: number
    Company_id?: number
    Company?: Company
    Product?: Product
}